.github/workflows/kernel-version-test.yml

```
@@ -0,0 +1,103 @@
+name: Kernel Version Test
+
+on:
+  workflow_call:
+    inputs:
+      kernel_version:
+        description: 'Version of the Linux kernel to build'
+        required: true
+        type: string
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    env:
+      KERNEL_VERSION: ${{ inputs.kernel_version }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install system dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y \
+            build-essential bison flex libssl-dev libelf-dev \
+            qemu-system-x86 busybox-static cpio xz-utils wget
+
+      - name: Cache Linux source
+        id: cache-kernel
+        uses: actions/cache@v4
+        with:
+          path: linux-${{ env.KERNEL_VERSION }}
+          key: kernel-${{ env.KERNEL_VERSION }}
+
+      - name: Download & build Linux kernel
+        if: steps.cache-kernel.outputs.cache-hit != 'true'
+        run: |
+          MAJOR=${KERNEL_VERSION%%.*}
+          wget https://cdn.kernel.org/pub/linux/kernel/v${MAJOR}.x/linux-${KERNEL_VERSION}.tar.xz
+          tar xf linux-${KERNEL_VERSION}.tar.xz
+          cd linux-${KERNEL_VERSION}
+          make defconfig
+          make -j$(nproc)
+
+      - name: Generate test binaries with io_uring enabled
+        run: |
+          # Build both integration (tokio/tests/) and unit (e.g., tokio/src/fs/file/tests.rs) tests with io_uring enabled
+          rustup target add x86_64-unknown-linux-musl
+          RUSTFLAGS="--cfg tokio_uring" \
+            cargo test -p tokio --features full \
+            --target x86_64-unknown-linux-musl --test 'fs*' --lib
```

```suggestion
          RUSTFLAGS="--cfg tokio_uring" \
            cargo build -p tokio --features full \
            --target x86_64-unknown-linux-musl --test 'fs*' --lib
```

`cargo test` also execute tests on the host machine, I'm not sure this is intentional, if not, we can use `cargo build`.